### PR TITLE
Add SourceResourceId optional param to Database API

### PR DESCRIPTION
**Pull request by imadan1996**
This PR makes the following changes: -

- The **important change** for this nuget version is the addition of the `SourceResourceId` optional input param to Database.cs.
- The changes made to DatabaseUpdate.cs are purely documentation related.
- Changes made to DatabaseOperations.cs, DatabaseOperationExtensions.cs, IDatabaseOperations.cs are documention related and re-ordering. Contents of these files have been re-ordered alphabetically, so that the next time someone re-generated these files locally that order will be preserved.

### CODE REVIEW

File: eng/mgmt/mgmtmetadata/sql_resource-manager.txt
Comment (imadan1996) [reply eng/mgmt/mgmtmetadata/sql_resource-manager.txt:14]: Yea.

Comment (imadan1996) [reply eng/mgmt/mgmtmetadata/sql_resource-manager.txt:14]: Should I revert it back to `v2`? All these files were auto generated.

Comment (ericshape) [reply eng/mgmt/mgmtmetadata/sql_resource-manager.txt:14]: if it is auto-assigned; it should be fine. just wait for the review from @fengzhou-msft

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**
